Source/WebCore/Modules/WebGPU/GPUCanvasContext.h

```diff
@@ -25,41 +25,71 @@
 
 #pragma once
 
+#include "GPUBasedCanvasRenderingContext.h"
+#include "GPUCanvasConfiguration.h"
 #include "GPUTexture.h"
 #include "HTMLCanvasElement.h"
 #include "OffscreenCanvas.h"
+#include "WebGPUSurface.h"
 #include <variant>
 #include <wtf/Ref.h>
 #include <wtf/RefCounted.h>
 #include <wtf/RefPtr.h>
 
 namespace WebCore {
 
-class GPUAdapter;
-struct GPUCanvasConfiguration;
-
-class GPUCanvasContext : public RefCounted<GPUCanvasContext> {
+class GPUCanvasContext final : public GPUBasedCanvasRenderingContext {
+    WTF_MAKE_ISO_ALLOCATED(GPUCanvasContext);
 public:
 #if ENABLE(OFFSCREEN_CANVAS)
     using CanvasType = std::variant<RefPtr<HTMLCanvasElement>, RefPtr<OffscreenCanvas>>;
 #else
     using CanvasType = std::variant<RefPtr<HTMLCanvasElement>>;
 #endif
 
-    static Ref<GPUCanvasContext> create()
-    {
-        return adoptRef(*new GPUCanvasContext());
-    }
+    static std::unique_ptr<GPUCanvasContext> create(CanvasBase&);
 
-    CanvasType canvas();
+    // CanvasRenderingContext overrides:
+    DestinationColorSpace colorSpace() const override;
+    bool compositingResultsNeedUpdating() const override { return m_compositingResultsNeedsUpdating; }
+    RefPtr<GraphicsLayerContentsDisplayDelegate> layerContentsDisplayDelegate() override;
+    bool needsPreparationForDisplay() const override { return true; }
+    void prepareForDisplay() override;
+    PixelFormat pixelFormat() const override;
+    void reshape(int width, int height) override;
+    // FIXME: implement these to allow for painting
+    void prepareForDisplayWithPaint() override { }
+    void paintRenderingResultsToCanvas() override { }
 
-    void configure(const GPUCanvasConfiguration&);
+    // GPUCanvasContext methods:
+    CanvasType canvas();
+    void configure(GPUCanvasConfiguration&&);
+    RefPtr<GPUTexture> getCurrentTexture();
     void unconfigure();
 
-    RefPtr<GPUTexture> getCurrentTexture();
+    bool isWebGPU() const override { return true; }
```

Comment:
Should be private

Reply:
If it's public in the parent then should it be public here too? Kinda no point if we make it private, we can just upcast and call it.